Source/Public/Get-SegmentedPath.ps1

```diff
@@ -41,49 +41,89 @@ function Get-SegmentedPath {
         [PoshCode.Pansies.RgbColor]$ErrorBackgroundColor
     )
 
-    $buffer = @()
+    # Define reusable constants.
+    $directorySeparatorChar = [System.IO.Path]::DirectorySeparatorChar
+    $ellipses = data { [char]0x2026 }
+    $resolvedHome = (Resolve-Path -LiteralPath $Home).Path
+    Write-Debug "`$directorySeparatorChar = '$directorySeparatorChar'"
+    Write-Debug "`$ellipses = '$ellipses'"
+    Write-Debug "`$resolvedHome = '$resolvedHome'"
```

Comment:
We don't want debug or verbose output in functions that are meant for use in your prompt, because your prompt runs when you're debugging. I know I had some in here -- it was always a bad idea.